### PR TITLE
Refactor WinsockBSD to better conform to interop guidelines.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
@davidsh, @stephentoub PTAL. Note that pgavlin@8e0c753 is the correct commit to look at--the other commits are part of #38.

**Comment by davidsh** [comment]
LGTM on https://github.com/pgavlin/corefx/commit/8e0c753

File: src/Common/src/Interop/Windows/Winsock/Interop.WSAConnect.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (CIPop) [inline src/Common/src/Interop/Windows/Winsock/Interop.WSAConnect.cs:1]: @pgavlin I wouldn't revert this change since it might actually be needed by NetworkInformation but splitting to a greater granularity isn't required:

> If multiple related DllImports will all be needed by every consumer, they may be declared in the same file, named for the functionality grouping, e.g. Interop.IOErrors.cs.
> (https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md)

File: src/Common/src/Interop/Windows/Winsock/Interop.WinsockAsync.cs [hunk 2]
```diff
@@ -34,9 +34,12 @@ internal struct ControlDataIPv6
         [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
         internal struct WSAPROTOCOLCHAIN
         {
-            internal int ChainLen;                                 /* the length of the chain,     */
+            // The length of the chain.
+            internal int ChainLen;
+
+            // A list of catalog entry IDs.
             [MarshalAs(UnmanagedType.ByValArray, SizeConst = 7)]
-            internal uint[] ChainEntries;       /* a list of dwCatalogEntryIds */
```
Comment (CIPop) [inline src/Common/src/Interop/Windows/Winsock/Interop.WinsockAsync.cs:39]: @stephentoub, @pgavlin  Are we considering ever changing these to the actual SDK names? If yes, I would prefer if we leave comments like the one above in - it makes search through the SDK much simpler.

File: src/Common/src/Interop/Windows/Winsock/Interop.WinsockAsync.cs [hunk 3]
```diff
@@ -184,213 +176,5 @@ internal struct WSAMsg
             internal WSABuffer controlBuffer;
             internal SocketFlags flags;
         }
-
```
Comment (CIPop) [inline src/Common/src/Interop/Windows/Winsock/Interop.WinsockAsync.cs:187]: According to the guideline it appears that this is not the organization that we're aiming for:

> If structs/constants will be used on their own without an associated DllImport, or if they may be used with multiple DllImports not in the same file, they should be declared in a separate file.

I don't think these structs can be used by themselves.

**Comment by CIPop** [comment]
Not sure if this respects the current corefx interop guidelines but I personally like it this way better. Thanks @pgavlin for doing this tedious work!

LGTM
